cmake/JoinPaths.cmake

```diff
@@ -0,0 +1,13 @@
+function(join_paths joined_path first_path_segment)
```

Comment:
This file should have a copyright and a license statement (copied from the source if it's not original to you).